includes/payment-methods/class-upe-payment-method.php

```diff
@@ -367,6 +367,91 @@ public function get_countries() {
 		return $this->has_domestic_transactions_restrictions() ? [ $account_country ] : $this->countries;
 	}
 
+	/**
+	 * Returns payment method description for the settings page.
+	 *
+	 * @param string|null $account_country Country of merchants account.
+	 *
+	 * @return string
+	 */
+	public function get_description( ?string $account_country = null ) {
+		if ( null !== $this->definition ) {
+			return $this->definition::get_description( $account_country );
+		}
+		return '';
+	}
+
+	/**
+	 * Returns payment method settings label.
+	 *
+	 * @param string $account_country Country of merchants account.
+	 * @return string
+	 */
+	public function get_settings_label( string $account_country ) {
+		if ( null !== $this->definition ) {
+			return $this->definition::get_settings_label( $account_country );
+		}
+		return $this->get_title( $account_country );
+	}
+
+	/**
+	 * Returns payment method settings icon.
+	 *
+	 * @param string|null $account_country Country of merchants account.
+	 * @return string
+	 */
+	public function get_settings_icon_url( ?string $account_country = null ) {
+		if ( null !== $this->definition ) {
+			return $this->definition::get_settings_icon_url( $account_country );
+		}
+		return $this->get_icon( $account_country );
+	}
+
+	/**
+	 * Returns boolean dependent on whether payment method allows manual capture.
+	 *
+	 * @return bool
+	 */
+	public function allows_manual_capture() {
+		if ( null !== $this->definition ) {
+			return $this->definition::allows_manual_capture();
+		}
+
+		return false;
+	}
+
+	/**
+	 * Returns the Stripe key for the payment method.
+	 *
+	 * @return string
+	 */
+	public function get_stripe_key() {
+		if ( null !== $this->definition ) {
+			return $this->definition::get_stripe_id();
+		}
+		return \WC_Payments::get_gateway()->get_payment_method_capability_key_map()[ $this->stripe_id ];
+	}
+
+	/**
+	 * Returns payment method settings definition.
+	 *
+	 * @param string $account_country Country of merchants account.
+	 * @return array
+	 */
+	public function get_payment_method_information_object( string $account_country ) {
+		return [
+			'id'                            => $this->stripe_id,
+			'label'                         => $this->get_settings_label( $account_country ),
```

Comment:
I like having a specific `get_settings_label()` method, but should we update the key in this object to `settings_label`? Obviously, that would require updating the consumers of this object, but I like the idea of having an object that contains the static details of a payment method. What do you think?

Or are we likely to never use a label other than the payment method title in practice and we can change it to `title`?

Reply:
> should we update the key in this object to settings_label

I also thought about this to make the property names more consistent but I didn't want to increase the scope of this PR so I decided to leave it out.

> Or are we likely to never use a label other than the payment method title in practice and we can change it to title?

I think we only need to perform this override for Credit Cards, which title is `Cards` but the label used in settings is `Credit / Debit Cards`.

Reply:
Ok. We can leave this to a future issue.